mlx/backend/common/quantized.cpp

```diff
@@ -22,13 +45,12 @@ void _qmm(
     int N,
     int K) {
   constexpr int bitmask = (1 << bits) - 1;
-  constexpr int pack_factor = 32 / bits;
+  constexpr int pack_factor = bits == 3 ? 8 : bits == 6 ? 4 : 8 / bits;
+  constexpr int bytes_per_pack = (bits == 3 || bits == 6) ? 3 : 1;
   constexpr int packs_in_group = group_size / pack_factor;
-  const int Ng = N / group_size;
-  const int Nw = N / pack_factor;
 
   for (int m = 0; m < M; m++) {
```

Comment:
I ran a microbench and seems like this doesn't affect the cpu `qmm` throughput:

```
python benchmarks/python/comparative/bench_mlx.py quant_matmul_t_32_4 --size 1x1x1024 --size 1x2048x128 --size 1x2048x32 --size 1x2048x32 --dtype float16 --dtype uint32 --dtype float16 --dtype float16 --cpu
```

Before: 6.2653ms
After: 6.2657ms